flink-ml-lib/src/main/java/org/apache/flink/ml/classification/knn/KnnParams.java

```diff
@@ -0,0 +1,38 @@
+package org.apache.flink.ml.classification.knn;
+
+import org.apache.flink.ml.param.IntParam;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+import org.apache.flink.ml.param.WithParams;
+import org.apache.flink.ml.params.shared.colname.HasFeatureColsDefaultAsNull;
+import org.apache.flink.ml.params.shared.colname.HasKnnDistanceType;
+import org.apache.flink.ml.params.shared.colname.HasLabelCol;
+import org.apache.flink.ml.params.shared.colname.HasPredictionCol;
+import org.apache.flink.ml.params.shared.colname.HasPredictionDetailCol;
+import org.apache.flink.ml.params.shared.colname.HasReservedColsDefaultAsNull;
+import org.apache.flink.ml.params.shared.colname.HasVectorColDefaultAsNull;
+
+/** knn fit parameters. */
+public interface KnnParams<T>
+        extends WithParams<T>,
+                HasVectorColDefaultAsNull<T>,
+                HasKnnDistanceType<T>,
+                HasLabelCol<T>,
+                HasFeatureColsDefaultAsNull<T>,
+                HasPredictionCol<T>,
+                HasPredictionDetailCol<T>,
```

Comment:
Spark does not support `HasPredictionDetailCol`. Any chance we can try to be consistent with Spark user experience for now? Otherwise, could you help explain the use-case and explain how this use-case is supported for Spark users?

Reply:
If user want to get two or three possible classification results, by setting HasDetail knn will give more classification results. then user can do more thing with the detail, just like recommendation. 

if user want to give only one classification result, he need not set this parameter.

Reply:
after discuss with lindong, I agree this suggestion.
I have remove it already.